code/obj/bookcase.dm

```diff
@@ -338,8 +338,102 @@
 	list(1,3,27,8),\
 	list(1,3,31,8))
 
-/obj/item/bookshelf_parts  //these arent finished, but i dont know how to do all the construction stuff, theres 4 bookshelves, standalone, middle, left endcap, right endcap.
+/obj/bookshelf/persistent
+	desc = "This bookshelf doesn't get cleaned out between shifts. Neat!"
+//these two make em look good on maps
+	pixel_y = 24
+	density = 0
+
+	New()
+		..()
+		START_TRACKING
+		src.load_old_books()
+
+	disposing()
+		STOP_TRACKING
+		..()
+
+	proc/load_old_books()
+		var/list/old_contents = list()
+		var/file_name = "data/persistent_bookshelf.json"
+		if (fexists(file_name))
+			old_contents = json_decode(file2text(file_name))
+			build_old_contents(old_contents)
+
+	proc/file_curr_books(var/list/curr_contents)
+		if (!curr_contents.len)
+			return
+		var/file_name = "data/persistent_bookshelf.json"
+		if(fexists(file_name))
+			fdel(file_name) //we rly dont want a duplicate, or to accidentally output twice to the file, it could screw the whole thing up
+		text2file(json_encode(curr_contents), file_name)
+
+	proc/build_old_contents(var/list/old_contents) //this goes and takes our giant weird list and makes it into books
+		if (old_contents.len)
+			for (var/i = 1, i <= old_contents.len, i++)
+				var/list/book_vars = old_contents[i]
```

Comment:
You could just do `for(var/list/book_vars in old_contents)` instead of these three lines.

Reply:
implemented this and pushed it